### PR TITLE
feat(reporting): Report when a mutant is in a static context

### PR DESCRIPTION
**Pull request by rouke-broersma**
With this change you should be able to see in the json, html and dashboard reporter that a mutation is in a static context

Also introduce some files to be able to display test information later on
And a little code cleanup

Related to #1565

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestCaseTests.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using Shouldly;
+using Stryker.Core.ProjectComponents.TestProjects;
+using Xunit;
+
+namespace Stryker.Core.UnitTest.ProjectComponents.TestProjects
+{
+    public class TestCaseTests
+    {
+        [Fact]
+        public void TestCaseEquals()
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestCaseTests.cs:11]:
```suggestion
        public void TestCasesWithEqualValuesShouldBeEqual()
```

File: src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestFileTests.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using Shouldly;
+using Stryker.Core.ProjectComponents.TestProjects;
+using Xunit;
+
+namespace Stryker.Core.UnitTest.ProjectComponents.TestProjects
+{
+    public class TestFileTests
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestFileTests.cs:9]: I'm missing tests for the Equality compare

File: src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestFileTests.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        [Fact]
+        public void MergeTestFiles()
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestFileTests.cs:12]:
```suggestion
        public void TestFilePlusTestFileShouldMerge()
```

File: src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestProjectTests.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+using Shouldly;
+using Stryker.Core.ProjectComponents.TestProjects;
+using Xunit;
+
+namespace Stryker.Core.UnitTest.ProjectComponents.TestProjects
+{
+    public class TestProjectTests
+    {
+        [Fact]
+        public void TestProjectEquals()
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestProjectTests.cs:14]:
```suggestion
        public void TestProjectsWithEqualValuesShouldBeEqual()
```

File: src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestProjectsInfoTests.cs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+using Shouldly;
+using Stryker.Core.ProjectComponents.TestProjects;
+using Xunit;
+
+namespace Stryker.Core.UnitTest.ProjectComponents.TestProjects
+{
+    public class TestProjectsInfoTests
+    {
+        [Fact]
+        public void MergeTestProjectsInfo()
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestProjectsInfoTests.cs:14]:
```suggestion
        public void ProjectsInfoPlusProjectsInfoShouldMerge()
```

File: src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestProjectsInfoTests.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+using System.Collections.Generic;
+using Buildalyzer;
+using Moq;
+using Shouldly;
+using Stryker.Core.ProjectComponents.TestProjects;
+using Xunit;
+
+namespace Stryker.Core.UnitTest.ProjectComponents.TestProjects
+{
+    public class TestProjectsInfoTests
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/ProjectComponents/TestProjects/TestProjectsInfoTests.cs:11]: I miss a test for the equals operator here

**Review by richardwerkman — CHANGES_REQUESTED** [review]
Missing two tests :)

**Comment by psfinaki** [comment]
Hmm I'm trying to get my head around that to understand if this anyhow affects us. This doesn't affect numbers and scores, just adds a marker in the report, right?
image

**Comment by rouke-broersma** [comment]
This just adds a marker in the report. There is behavior associated with this marker but that is not new.

**Comment by psfinaki** [comment]
Alright, thanks!

**Pull request merged**